### PR TITLE
feat: support JDBC connection and fix insert prepared statement

### LINKED ISSUE #81: [JDBC] Use url to init connection

### LINKED ISSUE #1496: jdbc: inserted rows are still in cache after `execute`
https://github.com/4paradigm/OpenMLDB/blob/90147ca4484ec40786c2df50faccb9bb7c31c0c2/java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/InsertPreparedStatementImpl.java#L361

We should delete currentRows after `ExecuteInsert`.
So that, we can do
```
while:
  stmt.setXX()
  stmt.execute()
```

### PR DESCRIPTION
**Pull request by vagetablechicken**
closes #81 , closes #1496

We can use `jdbc:openmldb:///<db>?zk=<zk_cluster>&zkPath=<zk_path>` to connect openmldb. The url schema is the same with python sqlalchemy.

### CODE REVIEW

**Comment by vagetablechicken** [comment]
[Apache Pulsar connector](https://trello.com/c/RFPkvEzM/42-apache-pulsar-connector)

File: java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/InsertPreparedStatementImpl.java [hunk 1]
```diff
@@ -84,6 +77,20 @@ public InsertPreparedStatementImpl(String db, String sql, SQLRouter router) thro
...
+        Status status = new Status();
+        SQLInsertRow row = router.GetInsertRow(db, sql, status);
+        if (status.getCode() != 0) {
+            String msg = status.getMsg();
+            status.delete();
+            if (row != null) {
+                row.delete();
+            }
+            throw new SQLException("getSQLInsertRow failed, " + msg);
+        }
```
Comment (dl239) [inline java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/InsertPreparedStatementImpl.java:90]: delete status

File: src/sdk/sql_insert_row.h [hunk 2]
```diff
@@ -125,10 +125,20 @@ class SQLInsertRows {
     inline uint32_t GetCnt() { return rows_.size(); }
     inline std::shared_ptr<SQLInsertRow> GetRow(uint32_t i) {
         if (i >= rows_.size()) {
-            return std::shared_ptr<SQLInsertRow>();
+            return {};
         }
         return rows_[i];
     }
+    inline const std::shared_ptr<hybridse::sdk::Schema> GetSchema() { return schema_; }
+    const std::vector<uint32_t> GetHoleIdx() {
```
Comment (dl239) [inline src/sdk/sql_insert_row.h:133]: Where are the method used

File: java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/InsertPreparedStatementImpl.java [hunk 3]
```diff
@@ -351,22 +349,26 @@ public void setObject(int i, Object o) throws SQLException {
...
             throw new SQLException("please use executeBatch");
         }
-        dataBuild();
+        buildRow();
         Status status = new Status();
-        boolean ok = router.ExecuteInsert(db, currentSql, currentRows, status);
+        // actually only one row
+        boolean ok = router.ExecuteInsert(db, sql, currentRows.get(0), status);
+        // cleanup rows even if insert failed
+        // we can't execute() again without set new row, so we must clean up here
```
Comment (dl239) [inline java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/InsertPreparedStatementImpl.java:371]: Check the InsertRow in currentRows whether need to delete explicitly.

File: src/sdk/sql_insert_row.h (hunk 2 above)
Comment (vagetablechicken) [reply src/sdk/sql_insert_row.h:133]: This change was for old insert prepared statement.
I leave it, cuz it's useful when we want get schema of rows. Before this change, we must create a new row to get. It's not good.

File: java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/InsertPreparedStatementImpl.java (hunk 3 above)
Comment (vagetablechicken) [reply java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/InsertPreparedStatementImpl.java:371]: Yes, it should be deleted, the same in close().

**Review by dl239 — APPROVED** [review]

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**